pennylane/devices/qutrit_mixed/measure.py

```diff
@@ -0,0 +1,309 @@
+        return math.reshape(summed_probs, new_shape)
+
+    return probs
+
+
+def calculate_variance(
+    measurementprocess: StateMeasurement, state: TensorLike, is_state_batched: bool = False
+) -> TensorLike:
+    """Find variance of observable.
+
+    Args:
+        measurementprocess (StateMeasurement): measurement to apply to the state
+        state (TensorLike): state to apply the measurement to
+        is_state_batched (bool): whether the state is batched or not
+
+    Returns:
+        TensorLike: the result of the measurement
+    """
+    probs = calculate_probability(measurementprocess, state, is_state_batched)
+    eigvals = math.asarray(measurementprocess.eigvals(), dtype="float64")
+    # In case of broadcasting, `probs` has two axes and these are a matrix-vector products
+    return math.dot(probs, (eigvals**2)) - math.dot(probs, eigvals) ** 2
+
+
+# pylint: disable=too-many-return-statements
+def get_measurement_function(
+    measurementprocess: MeasurementProcess,
+    state: TensorLike,  # pylint: disable=unused-argument
+) -> Callable[[MeasurementProcess, TensorLike], TensorLike]:
+    """Get the appropriate method for performing a measurement.
+
+    Args:
+        measurementprocess (MeasurementProcess): measurement process to apply to the state
+        state (TensorLike): the state to measure
+        is_state_batched (bool): whether the state is batched or not
+
+    Returns:
+        Callable: function that returns the measurement result
+    """
+    if isinstance(measurementprocess, StateMeasurement):
+        if isinstance(measurementprocess, ExpectationMP):
+            # TODO add faster methods
+            # TODO add support for sparce Hamiltonians
+            if isinstance(measurementprocess.obs, Hamiltonian):
+                return sum_of_terms_method
+            if isinstance(measurementprocess.obs, Sum):
+                return sum_of_terms_method
+            if measurementprocess.obs.has_matrix:
+                return trace_method
+        if measurementprocess.obs is None or measurementprocess.obs.has_diagonalizing_gates:
+            if isinstance(measurementprocess, StateMP):
+                return reduce_density_matrix
+            if isinstance(measurementprocess, ProbabilityMP):
+                return calculate_probability
+            if isinstance(measurementprocess, VarianceMP):
+                return calculate_variance
+
+    raise NotImplementedError
+
+
+def measure(
+    measurementprocess: MeasurementProcess, state: TensorLike, is_state_batched: bool = False
+) -> TensorLike:
+    """Apply a measurement process to a state.
+
+    Args:
+        measurementprocess (MeasurementProcess): measurement process to apply to the state
+        state (TensorLike): the state to measure
+        is_state_batched (bool): whether the state is batched or not
+
+    Returns:
+        Tensorlike: the result of the measurement
+    """
+    return get_measurement_function(measurementprocess, state)(
```

Comment:
What is the rationale for having `get_measurement_function` defined separately instead of having the same logic in here?

Reply:
To remain consistant with twhat was done with `default.qubit` and to make it separately callable in a test. It could be reformatted if you prefer.

Reply:
We did so in default qubit to better separate the logic for determining what method we want to use from the logic for those methods themselves. This allows better unit testing the choice of kernel without resorting to mockers and spying.

It's convenient for `default.qubit`, as we have four different kernels that we choose between based on a variety of constraints and optimizations. 

But that might premature optimization and speculative generality here.

Reply:
So would you like that to be removed?

Reply:
Let's leave it as-is, for ease of testing and consistency.